source/postprocess/melt_statistics.cc

```diff
@@ -0,0 +1,182 @@
+/*
+  Copyright (C) 2011 - 2015 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+
+#include <aspect/postprocess/melt_statistics.h>
+#include <aspect/simulator_access.h>
+#include <aspect/melt.h>
+
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/fe/fe_values.h>
+
+#include <boost/archive/text_oarchive.hpp>
+#include <boost/archive/text_iarchive.hpp>
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    template <int dim>
+    std::pair<std::string,std::string>
+    MeltStatistics<dim>::execute (TableHandler &statistics)
+    {
+      // create a quadrature formula based on the temperature element alone.
```

Comment:
We probably make this mistake in many places :-)

On the positive side, the quadrature formulas are chosen as appropriate for the matrix terms, which have twice the polynomial degree as the fields themselves. On the other hand, most of the time you are integrating something that only includes each field once; if the quantity you compute is a linear function of, say, the temperature, then your integrand has only half the polynomial degree as in the integrand of the matrix. If you compute something nonlinear of the temperature/composition/etc, then that argument is not valid, but you're likely still doing more than actually necessary.